frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,259 @@ pub struct PoolRoles<AccountId> {
 	pub state_toggler: Option<AccountId>,
 }
 
+/// Pool commission.
+///
+/// The pool `root` can set commission configuration after pool creation. By default, all commission
+/// values are `None`. Pool `root`s can set `max` and `throttle` configurations before setting an
+/// initial commission `current` value - the commission percentage and payee of commission.
+///
+/// The `max` commission value can only be decreased after the initial value is
+/// set, to prevent commission from repeatedly increasing.
+///
+/// An optional commission throttle allows the pool to set strict limits to how much
+/// commission can change in each update, and how often updates can take place.
+/// If a `throttle` is set before a commission `current` is set, the initial commission `current`
+/// value will not be subject to throttling. Subsequent commission updates will be.
+#[derive(
+	Encode, Decode, DefaultNoBound, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value
+	/// can only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configuration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub throttle: Option<CommissionThrottle<T>>,
+}
+
+impl<T: Config> Commission<T> {
+	/// Get the current commission percentage of this pool.
+	///
+	/// Returns zero if commission has not yet been configured.
+	fn commission_or_zero(&self) -> Perbill {
+		self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero())
+	}
+
+	/// Returns true if a commission percentage updating to `to` would exhaust the throttle limit.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the previous commission update took place, and
+	/// 2. the new commission is larger than the maximum allowed increase.
+	///
+	/// Throttling is not applied to commission updates if `current` is still `None`.
+	fn throttling(&self, to: &Perbill) -> bool {
+		if let Some(t) = self.throttle.as_ref() {
+			// factor previously updated block into whether user is throttled.
+			if t.previous_set_at.map_or(
+				// if no `previous_set_at` is set, throttled if the attempted increase
+				// in commission is greater than `max_increase`.
+				(*to).saturating_sub(self.commission_or_zero()) > t.change_rate.max_increase,
```

Comment:
Given `current` commission = 5%, when a pool operator sets throttle for the first time (at say block 10) with `max_increase`=1% and `min_delay` = 2 blocks, and at block 16 tries to increase `current` commission to 7% (6 blocks passed so `max_allowed_increase` = 3%, `attempted_increase`=2%), it would still fail since we do not keep the `previous_set_at` and only compare to `max_increase`=1% for the first update after setting the throttle. 

As a user I would not expect that. We should probably initialise `previous_set_at` when `throttling` is set for the first time. Its cheating a bit but I feel is still better than  keeping `None`. 
Even better to always update `previous_set_at` when `commission` is set irrespective of throttle (and potentially decouple it from throttle).

Reply:
This is the concern I had before, but the solution would be to have a maximum commission set so it becomes a bit more predictable.

Reply:
> As a user I would not expect that. We should probably initialise previous_set_at when throttling is set for the first time. Its cheating a bit but I feel is still better than keeping None.

I was debating this too, decoupling `previous_set_at` would simplify things and remove the need for 2 structs for Throttle and ChangeRate. I also see a thumbs up from @kianenigma, so will go ahead and try to resolve this.

Reply:
`previous_set_at` is now a `last_updated` field below `current`, and is updated on every commission change. This also removed the need for the `CommissionThrottle` struct - we go straight to the `CommissionChangeRate` struct now.

It is now also registering an update if we update `max` commission to a lower value than current commission. This happens in `maybe_update_max`.